src/Elliptic/DiscontinuousGalerkin/SubdomainOperator/InitializeSubdomain.hpp

```diff
@@ -0,0 +1,521 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/range/join.hpp>
+#include <cstddef>
+#include <tuple>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/SliceVariables.hpp"
+#include "DataStructures/Tensor/EagerMath/Determinant.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/CreateInitialElement.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/InterfaceHelpers.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Structure/CreateInitialMesh.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/DirectionMap.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/IndexToSliceAt.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/DiscontinuousGalerkin/SubdomainOperator/Tags.hpp"
+#include "Elliptic/Utilities/ApplyAt.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+struct GlobalCache;
+}  // namespace Parallel
+namespace tuples {
+template <typename... Tags>
+struct TaggedTuple;
+}  // namespace tuples
+/// \endcond
+
+/// Actions related to the DG subdomain operator
+namespace elliptic::dg::subdomain_operator::Actions {
+
+namespace detail {
+// Initialize the geometry of a neighbor into which an overlap extends
+template <size_t Dim>
+void initialize_overlap_geometry(
```

Comment:
I didn't do that so far because it would "triplicate" the lengthy function signature (definition + template instantiation). I'll merge this code with `InitializeDomain` in an upcoming PR, which is currently being prepared by #3217, so my suggestion is to keep this in the header until then.

Reply:
I'm inclined to encourage even verbose function signatures be cpp+explicit instantiation, because it's good policy for managing compilation time -- code present in headers is dragged into everything with the header in the include chain, where cpp code gets compiled only for this library and linked into the rest.

I concede it boosts the line count here, but it's not terribly hard to read the duplicate signatures. Unless you feel very strongly about it, I think it's best to move it to the cpp -- presumably even when it gets merged there will be a similar subroutine that should end up in the cpp regardless, right?

Reply:
I pushed the remaining preparation to merge the initialization routines to a separate PR (#3247)